src/net.cpp

```diff
@@ -1398,6 +1403,16 @@ void CConnman::ThreadSocketHandler()
                 }
             }
         }
+        //
+        // Reduce number of connections, if needed
+        //
+        if (vNodesCopy.size() > (size_t)nMaxConnections)
+        {
```

Comment:
I think it would be good to have a debug message here, so that it's possible to see that this is being triggered. Something like:

```suggestion
        {
            LogPrint("net", "%s: attempting to reduce connections: max=%u current=%u", __func__, vNodesCopy.size(), nMaxConnections);
```

Maybe that should even be LogPrintf and part of the default output rather than only in debug mode.